client/me/purchases/remove-plan-dialog/index.tsx

```diff
@@ -0,0 +1,162 @@
+import FormattedHeader from 'calypso/components/formatted-header';
+import { SiteExcerptData } from 'calypso/data/sites/site-excerpt-types';
+import { SiteScreenshot } from '../site-screenshot/';
+import getPlanFeatures from './get-plan-features';
+
+import './style.scss';
+
+interface RemovePlanDialogProps {
+	closeDialog: () => void;
+	removePlan: () => void;
+	isDialogVisible: boolean;
+	isRemoving: boolean;
+	site: SiteExcerptData;
+	hasDomain: boolean;
+	wpcomSiteURL: string;
+}
+
+export const RemovePlanDialog = ( {
+	closeDialog,
+	removePlan,
+	isDialogVisible,
+	site,
+	hasDomain,
+	wpcomSiteURL,
+}: RemovePlanDialogProps ) => {
+	const translate = useTranslate();
+	const siteName = site.name ?? '';
+
+	/**
+	 * Dialog buttons
+	 */
+	const buttons = [
+		{
+			action: 'remove',
+			label: translate( 'Cancel my plan' ),
+			onClick: removePlan,
+		},
+		{
+			action: 'cancel',
+			label: translate( 'Keep my plan' ),
+			onClick: closeDialog,
+			isPrimary: true,
+		},
+	];
+
+	/**
+	 * Istantiate site's plan variables.
+	 */
+	const productSlug = site.plan?.product_slug;
+	const planLabel = site.plan?.product_name_short;
+	const isComingSoon = site.is_coming_soon;
+	const isPrivate = site.is_private;
+	const launchedStatus = site.launch_status === 'launched' ? true : false;
+	const shouldUseSiteThumbnail =
+		isComingSoon === false && isPrivate === false && launchedStatus === true;
+	const subTitle =
+		hasDomain && wpcomSiteURL
+			? translate( 'If you cancel your plan, once it expires, you will lose:' )
+			: translate( 'If you cancel your plan, you will lose:' );
+
+	/**
+	 * Return the list of features that the user will lose by canceling their plan.
+	 *
+	 * @returns Fragment
+	 */
+	const FeaturesList = () => {
+		if ( typeof productSlug === 'string' ) {
+			const planFeatures = getPlanFeatures( productSlug, hasDomain, wpcomSiteURL );
+
+			if ( planFeatures.length > 0 ) {
+				const domainFeature =
+					hasDomain && wpcomSiteURL ? (
+						<p>
+							{ translate(
+								'Your custom domain as primary. Your traffic will be redirected to %(domain)s',
+								{
+									args: {
+										domain: wpcomSiteURL,
+									},
+								}
+							) }
+						</p>
+					) : null;
+				return (
+					<Fragment>
+						<p>{ subTitle }</p>
+						<ul className="remove-plan-dialog__list-plan-features">
+							{ domainFeature }
+							{ planFeatures.map( ( feature, index ) => {
+								return (
+									<li key={ index }>
+										<Gridicon
+											className="remove-plan-dialog__item-cross-small"
+											size={ 24 }
+											icon="cross-small"
+										/>
+										{ feature }
+									</li>
+								);
+							} ) }
+						</ul>
+					</Fragment>
+				);
+			}
+		}
+
+		return null;
+	};
+
+	/**
+	 * Dialog classname
+	 */
+	const dialogClassName = shouldUseSiteThumbnail
+		? 'remove-plan-dialog --with-screenshot'
+		: 'remove-plan-dialog';
+
+	/**
+	 * Plan cancellation dialog.
+	 */
+	return (
+		<Dialog
```

Comment:
This is still missing Tracks events: https://github.com/Automattic/wp-calypso/pull/67090#discussion_r965473304

Reply:
@michaeldcain could you please verify the tracking events added here https://github.com/Automattic/wp-calypso/pull/68067/files/80fea556de0d33e45084d1132c5d68428240734b#diff-22f6c96a70e669c1ee86915b052fe607759a7d300568afe8a92d9646c37860d0R33 ?

Reply:
@michaeldcain could you please reviews the changes above?

Reply:
@ciprianimike: sorry I missed this ping. Those events look good to me.